### DISCUSSION TITLE
Is it possible to separate your caching system to a standalone library?

### DISCUSSION THREAD

**Discussion by LowerDeez** [OpenSlides/OpenSlides#6080] [General]
Hey! Is it possible to separate your caching system to a standalone library?
The problem of cache dependencies solved in an interesting way.

**Comment by tsiegleauq** [comment]
what cache specifically?

**Reply by LowerDeez** [reply]
So, I talk about these:
1. A way to build an object's full data to cache: https://github.com/OpenSlides/OpenSlides/blob/master/server/openslides/utils/models.py#L120
2. Caching logic: https://github.com/OpenSlides/OpenSlides/blob/master/server/openslides/utils/cache.py#L43
3. Redis cache provider: https://github.com/OpenSlides/OpenSlides/blob/master/server/openslides/utils/cache_providers.py#L115
4. Lock to avoid dogpile effect: https://github.com/OpenSlides/OpenSlides/blob/master/server/openslides/utils/locking.py
5. Autoupdate system to invalidate cache (some kind of observer?): https://github.com/OpenSlides/OpenSlides/blob/master/server/openslides/utils/autoupdate.py
6. Redis related staff: https://github.com/OpenSlides/OpenSlides/blob/master/server/openslides/utils/redis_connection_pool.py, https://github.com/OpenSlides/OpenSlides/blob/master/server/openslides/utils/redis.py

**Reply by tsiegleauq** [reply]
it's under the MIT license, you may create your own library out of these. I doubt anyone of us will ever do so.

**Reply by LowerDeez** [reply]
> it's under the MIT license, you may create your own library out of these. I doubt anyone of us will ever do so.

Thanks)
